### PR TITLE
Module for translation of Go modules to URLs

### PR DESCRIPTION
**Pull request by msehnout**
We want a function that takes whatever `go get` takes and translate it to some URL, which we can then use in the monitor. In the initial phase, we are only interested in modules backed by Github repositories.

Relevant docs:
https://golang.org/cmd/go/#hdr-Remote_import_paths
Relevant source code:
https://github.com/golang/go/blob/master/src/cmd/go/internal/get/get.go

### CODE REVIEW

**Comment by tisnik** [comment]
@msehnout could you please setup CI jobs? Pls don't hesitate to ask me if you need help with it :)

File: gopkg/translate.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+def translate(pkg: str) -> str:
```
Comment (tisnik) [inline gopkg/translate.py]: are we planning to use types? @msrb? please make sure the CI will use the right Python version then

Comment (msehnout) [reply gopkg/translate.py]: I decided to leave them out.

File: gopkg/translate.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            except KeyError:
+                pass
+
+    return None
+
+
+def translate(pkg):
+    """
+    Takes whatever `go get` takes and return string with organization/user and repository name.
+    It returns none for anything that is not available on Github.
```
Comment (tisnik) [inline gopkg/translate.py]: none/None

File: gopkg/translate.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                    return '{}/{}'.format(m.group('user'), m.group('repo'))
+            except KeyError:
+                pass
+
+    return None
+
+
+def translate(pkg):
+    """
+    Takes whatever `go get` takes and return string with organization/user and repository name.
```
Comment (tisnik) [inline gopkg/translate.py]: You'd need to use 'Take whatever....' to make pydoc style checker happy :)

File: gopkg/translate.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    m = GITHUB_REPO_RE.match(pkg)
+    if m is not None:
+        return '{}/{}'.format(m.group('user'), m.group('repo'))
+
+    # Test if the pkg name can be translated into Github repo
+    repo = get_repo_from_random_urn(pkg)
+    if repo is not None:
+        return repo
+
+    return None
```
Comment (tisnik) [inline gopkg/translate.py]: I'm not sure I understand the logic there ;-) Do you need the `if` at all? Looks like a leftover...

**Review by tisnik — APPROVED** [review]
LGTM so far, TYVM @msehnout. I just can't resist and add minor comments ;)

File: gopkg/translate.py (hunk 4 above)
Comment (msehnout) [reply gopkg/translate.py]: It is kind of a leftover. I thought there will be more cases than these two, so I used the pattern:
```
try it
if not none:
   return it
```
But there is no other case, so it does not make any sense for this one...

**Review by tisnik — APPROVED** [review]
LGTM, thanks a lot @msehnout

**Comment by tisnik** [comment]
feel free to merge it in current state please

**Pull request merged**